qiskit/circuit/library/generalized_gates/unitary.py

```diff
@@ -97,7 +100,7 @@ def __init__(
         # Convert to numpy array in case not already an array
         data = numpy.asarray(data, dtype=complex)
         input_dim, output_dim = data.shape
-        num_qubits = int(math.log2(input_dim))
+        num_qubits = num_qubits if num_qubits is not None else int(math.log2(input_dim))
```

Comment:
This `num_qubits` optimization seems a bit unecessary (30ns vs 120ns on my machine, but fine).

Reply:
That's fair - I did this when it was still `np.log2` and it was a huge overhead, and I didn't really think to switch it from the Numpy overhead at the time.